python/tskit/trees.py

```diff
@@ -1564,13 +1564,83 @@ def _levelorder_traversal(self, u):
+        """
         yield from sorted(
-            self.nodes(u, order="levelorder"), key=self.time, reverse=True
+            self.nodes(u, order="levelorder"),
+            key=lambda u: (self.time(u), u),
+            reverse=True,
         )
 
+    def _minlex_postorder_traversal(self, u):
+        """
+        Postorder traversal that visits leaves in minimum lexicographic order.
+
+        Minlex stands for minimum lexicographic. We wish to visit a tree in such
+        a way that the leaves visited, when their IDs are listed out, have
+        minimum lexicographic order. This is a useful ordering for drawing
+        multiple Trees of a TreeSequence, as it leads to more consistency
+        between adjacent Trees.
+        """
+        # We skip perf optimisations here (compared to _preorder_traversal and
+        # _postorder_traversal) as this ordering is unlikely to be used in perf
+        # sensitive applications
+        stack = collections.deque([u])
+        parent = NULL
+
+        # We compute a dictionary mapping from internal node ID to min leaf ID
+        # under the node, using a first postorder traversal
+        min_leaf_dict = {}
+        while len(stack) > 0:
+            v = stack[-1]
+            children = [] if v == parent else self.children(v)
+            if children:
+                # The first time visiting a node, we push its children onto the stack.
+                # reversed is not strictly necessary, but it gives the postorder
+                # we would intuitively expect.
+                stack.extend(reversed(children))
+            else:
+                # The second time visiting a node, we record its min leaf ID
+                # underneath, pop it, and update the parent variable
+                if v != parent:
+                    # at a leaf node
+                    min_leaf_dict[v] = v
+                else:
+                    # at a parent after finishing all its children
+                    min_leaf_dict[v] = min([min_leaf_dict[c] for c in self.children(v)])
+                parent = self.get_parent(v)
+                stack.pop()
+
+        # Now we do a second postorder traversal
+        stack.clear()
+        stack.extend([u])
+        parent = NULL
+        while len(stack) > 0:
+            v = stack[-1]
+            children = [] if v == parent else self.children(v)
+            if children:
+                # The first time visiting a node, we push onto the stack its children
+                # in order of reverse min leaf ID under each child. This guarantees
+                # that the earlier children visited have smaller min leaf ID,
+                # which is equivalent to the minlex condition.
+                stack.extend(
+                    sorted(children, key=lambda u: min_leaf_dict[u], reverse=True)
+                )
+            else:
+                # The second time visiting a node, we pop and yield it, and
+                # we update the parent variable
```

Comment:
Note the copious comments that have been added to this function

Reply:
Very nice, thanks!